ssl/record/ssl3_record_tls13.c

```diff
@@ -53,21 +53,27 @@ int tls13_enc(SSL *s, SSL3_RECORD *recs, size_t n_recs, int send)
     }
     ivlen = EVP_CIPHER_CTX_iv_length(ctx);
 
+    if (EVP_CIPHER_CTX_mode(ctx) == EVP_CIPH_CCM_MODE) {
+        if (s->s3->tmp.new_cipher->algorithm_enc
+                & (SSL_AES128CCM8 | SSL_AES256CCM8))
+            taglen = EVP_CCM8_TLS_TAG_LEN;
+         else
+            taglen = EVP_CCM_TLS_TAG_LEN;
+         if (send && EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_SET_TAG, taglen,
+                                         NULL) <= 0)
+            return -1;
+    } else {
+        taglen = EVP_GCM_TLS_TAG_LEN;
+    }
+
     if (!send) {
         /*
          * Take off tag. There must be at least one byte of content type as
          * well as the tag
          */
-        /*
-         * TODO(TLS1.3): We're going to need to figure out the tag len based on
-         * the cipher. For now we just support GCM tags.
-         * TODO(TLS1.3): When we've swapped over the record layer to TLSv1.3
-         * then the length must be 1 + the tag len to account for the content
-         * byte that we know must have been encrypted.
-         */
-        if (rec->length < EVP_GCM_TLS_TAG_LEN)
+        if (rec->length < taglen)
```

Comment:
Should this now be `if (rec->length < taglen + 1)` to account for the second TODO item you removed here? We've done the work to swap over the record layer.